lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -555,38 +556,58 @@ noExtraTxBodyContent = ExtraTxBodyContent [] [] [] id
 -- To avoid the need for `ledger -> wallet` conversions, this function can only
 -- be used to *add* tx body content.
 updateSealedTx
-    :: ExtraTxBodyContent
+    :: TxUpdate
     -> SealedTx
     -> Either ErrUpdateSealedTx SealedTx
 updateSealedTx extraContent (cardanoTx -> InAnyCardanoEra _era tx) = do
 
     -- NOTE: The script witnesses are carried along with the cardano-api
     -- `anyEraBody`.
     let (Cardano.Tx anyEraBody existingKeyWits) = tx
-    body' <- modifyLedgerTxBody extraContent anyEraBody
+    body' <- modifyLedgerTx extraContent anyEraBody
 
     unless (null existingKeyWits) $
        Left $ ErrExistingKeyWitnesses $ length existingKeyWits
 
     return $ sealedTxFromCardanoBody body'
 
   where
-    modifyLedgerTxBody
-        :: ExtraTxBodyContent
+    modifyLedgerTx
+        :: forall era. ()
+        => TxUpdate
         -> Cardano.TxBody era
         -> Either ErrUpdateSealedTx (Cardano.TxBody era)
-    modifyLedgerTxBody ebc (Cardano.ShelleyTxBody shelleyEra bod scripts scriptData aux val) =
-            Right $ Cardano.ShelleyTxBody shelleyEra (adjust ebc shelleyEra bod) scripts scriptData aux val
+    modifyLedgerTx ebc (Cardano.ShelleyTxBody shelleyEra bod scripts scriptData aux val) =
```

Comment:
Re the name change… It still takes a `TxBody` 🤷‍♂️

Reply:
The function is actually called `modifyLedgerTx` but it's not a "ledger tx" (if reading this as a tx from the cardano-ledger-specs), it's a cardano-api's tx.